Assets/Packages/com.studio23.ss2.upmassistant/Editor/Manager/DataHandler.cs

```diff
@@ -0,0 +1,253 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+using System.IO;
+using Studio23.SS2.UPMAssistant.Editor.Data;
+
+namespace Studio23.SS2.UPMAssistant.Editor
+{
+    
+ 
+public static class DataHandler
+{
+   public static readonly string ROOT = "Assets/Packages/";
+    public static readonly string DATA_PATH = "Assets/UPMAssistant/Editor/Data";
+    public static readonly string LICENSE_URL = "licenseURLData.json";
+    public static readonly string PACKAGE_NAME = "packageNameData.json";
+   
+    public const string DefaultPackageName = "com.[companyname].[project].[packagename]";
+    public const string DefaultLicenceURL = "https://api.github.com/licenses/mit";//"https://api.github.com/licenses/lgpl-2.1";
+    
+    public static readonly string PACKAGE_JSON = "package.json";
+    public static readonly string README_MD = "README.md";
+    public static readonly string CHANGELOG_MD = "CHANGELOG.md";
+    public static readonly string LICENSE_MD = "LICENSE.md";
+    public static readonly string THIRDPARTYNOTICES_MD = "ThirdPartyNotices.md";
+     
+    
+    public static Dictionary<string, bool> FolderAndFilesList = new Dictionary<string, bool>
+    {
+        {PACKAGE_JSON, false},
+        {README_MD, false},
+        {CHANGELOG_MD, false},
+        {LICENSE_MD, false},
+        {THIRDPARTYNOTICES_MD, false},
+        {"Editor", false},
+        {$"Editor/[[packagename]].editor.asmdef", false},
+        {"Runtime", false},
+        {$"Runtime/[[packagename]].asmdef", false},
+        {"Runtime/Data", false},
+        {"Runtime/Core", false},
+        {"Tests", false},
+        {"Tests/EditMode", false},
+        {$"Tests/EditMode/editmode.tests.asmdef", false},
+        {"Tests/PlayMode", false},
+        {$"Tests/PlayMode/playmode.tests.asmdef", false},
+        {"Samples", false},
+        {"Documentation", false},
+           
+    };
+    public static void Initialized()
+    {
+        if (!Directory.Exists(ROOT))
+        {
+            Directory.CreateDirectory(ROOT);
+            Debug.Log("Folder structure created: " + ROOT);
+        }
+        if (!Directory.Exists(DATA_PATH))
+        {
+            Directory.CreateDirectory(DATA_PATH);
+            Debug.Log("Folder structure created: " + DATA_PATH);
+        } 
+        
+        var licensePath = Path.Combine(DATA_PATH, LICENSE_URL);
+        if (!File.Exists(licensePath))
+        {
+            LicenseURLData data = new LicenseURLData { LicenseURL = DefaultLicenceURL };
+            string jsonData = JsonUtility.ToJson(data);
+            File.WriteAllText(licensePath, jsonData);
```

Comment:
async method should be used

Reply:
The File.WriteAllText method is a synchronous method, which means that it blocks the calling thread until the write operation is complete